#Lab 2


## Loading my Own Data 

In [1]:
import warnings
warnings.simplefilter('ignore')

import pandas as pd

df = pd.read_csv('s3://ia340-steinfadt/Vegetable-Prices-2022.csv') # load the data from s3
df.head(10) #show the top 5 rows

Vegetable    Form  RetailPrice RetailPriceUnit   Yield  \
0  Acorn squash   Fresh       1.2136       per pound  0.4586   
1     Artichoke   Fresh       2.4703       per pound  0.3750   
2     Artichoke  Canned       3.4498       per pound  0.6500   
3     Asparagus   Fresh       2.9531       per pound  0.4938   
4     Asparagus  Canned       3.4328       per pound  0.6500   
5     Asparagus  Frozen       6.8212       per pound  1.0335   
6      Avocados   Fresh       2.6737       per pound  0.7408   
7         Beets  Canned       1.1431       per pound  0.6500   
8   Black beans  Canned       1.2387       per pound  0.6500   
9   Black beans   Dried       1.5250       per pound  2.4692   

   CupEquivalentSize CupEquivalentUnit  CupEquivalentPrice  
0             0.4519            pounds              1.1961  
1             0.3858            pounds              2.5415  
2             0.3858            pounds              2.0476  
3             0.3968            pounds              2.3731  
4             0.3968            pounds              2.0958  
5             0.3968            pounds              2.6191  
6             0.3197            pounds              1.1538  
7             0.3748            pounds              0.6591  
8             0.3858            pounds              0.7352  
9             0.3858            pounds              0.2383

## View the Data

In [2]:
df.shape

(93, 8)

My data has 93 rows and 8 differnt columns

In [3]:
df.info

<bound method DataFrame.info of         Vegetable    Form  RetailPrice RetailPriceUnit   Yield  \
0    Acorn squash   Fresh       1.2136       per pound  0.4586   
1       Artichoke   Fresh       2.4703       per pound  0.3750   
2       Artichoke  Canned       3.4498       per pound  0.6500   
3       Asparagus   Fresh       2.9531       per pound  0.4938   
4       Asparagus  Canned       3.4328       per pound  0.6500   
..            ...     ...          ...             ...     ...   
88       Tomatoes  Canned       1.0715       per pound  1.0000   
89  Turnip greens   Fresh       2.9250       per pound  0.7500   
90  Turnip greens  Canned       1.3154       per pound  0.6500   
91  Turnip greens  Frozen       2.5169       per pound  0.7760   
92       Zucchini   Fresh       1.6359       per pound  0.7695   

    CupEquivalentSize CupEquivalentUnit  CupEquivalentPrice  
0              0.4519            pounds              1.1961  
1              0.3858            pounds            

In [5]:
df.describe

<bound method NDFrame.describe of         Vegetable    Form  RetailPrice RetailPriceUnit   Yield  \
0    Acorn squash   Fresh       1.2136       per pound  0.4586   
1       Artichoke   Fresh       2.4703       per pound  0.3750   
2       Artichoke  Canned       3.4498       per pound  0.6500   
3       Asparagus   Fresh       2.9531       per pound  0.4938   
4       Asparagus  Canned       3.4328       per pound  0.6500   
..            ...     ...          ...             ...     ...   
88       Tomatoes  Canned       1.0715       per pound  1.0000   
89  Turnip greens   Fresh       2.9250       per pound  0.7500   
90  Turnip greens  Canned       1.3154       per pound  0.6500   
91  Turnip greens  Frozen       2.5169       per pound  0.7760   
92       Zucchini   Fresh       1.6359       per pound  0.7695   

    CupEquivalentSize CupEquivalentUnit  CupEquivalentPrice  
0              0.4519            pounds              1.1961  
1              0.3858            pounds          

Within the 8 columns: there are 4 number type columns and 4 text columns

## Clean the Data

In [6]:
#renames two columns
df = df.rename(columns={"Vegetable": "Veggie", "RetailPrice": "RP"})
df.head()

Veggie    Form      RP RetailPriceUnit   Yield  CupEquivalentSize  \
0  Acorn squash   Fresh  1.2136       per pound  0.4586             0.4519   
1     Artichoke   Fresh  2.4703       per pound  0.3750             0.3858   
2     Artichoke  Canned  3.4498       per pound  0.6500             0.3858   
3     Asparagus   Fresh  2.9531       per pound  0.4938             0.3968   
4     Asparagus  Canned  3.4328       per pound  0.6500             0.3968   

  CupEquivalentUnit  CupEquivalentPrice  
0            pounds              1.1961  
1            pounds              2.5415  
2            pounds              2.0476  
3            pounds              2.3731  
4            pounds              2.0958

In [7]:
df.isnull().sum() #show the number of rows containing null value

Veggie                0
Form                  0
RP                    0
RetailPriceUnit       0
Yield                 0
CupEquivalentSize     0
CupEquivalentUnit     0
CupEquivalentPrice    0
dtype: int64

- I changed two of the colunm names "Vegetable and "RetailPrice"
- Currently there are no Null values from checking so nothing to fix here. if there any you use "df = df.dropna()" to remove them

## Manipulate the Data

In [8]:
df.describe()

RP      Yield  CupEquivalentSize  CupEquivalentPrice
count  93.000000  93.000000          93.000000           93.000000
mean    2.107941   0.959997           0.338635            0.828323
std     1.082376   0.494943           0.066544            0.508993
min     0.797000   0.375000           0.154300            0.221500
25%     1.269100   0.650000           0.297600            0.539700
50%     1.843300   0.881800           0.341700            0.701800
75%     2.494100   0.970200           0.385800            0.937300
max     6.821200   2.539700           0.540100            2.619100

In average the reail price of veggies is $2.10 but the cheapest is 79 cents where as the most expensive one is $6.82 which is a significant gap between veggie types

In [12]:
df['Veggie']

0      Acorn squash
1         Artichoke
2         Artichoke
3         Asparagus
4         Asparagus
          ...      
88         Tomatoes
89    Turnip greens
90    Turnip greens
91    Turnip greens
92         Zucchini
Name: Veggie, Length: 93, dtype: object

In [13]:
df.sort_values(by='Form') #sort data by a column

Veggie    Form      RP  \
24                                           Carrots  Canned  1.1979   
44                                        Green peas  Canned  1.3118   
42                                       Green beans  Canned  1.1291   
55                                        Lima beans  Canned  1.4785   
39                              Great northern beans  Canned  1.1583   
..                                               ...     ...     ...   
61  Mixed vegetables, broccoli, cauliflower, carrots  Frozen  2.1538   
36                                              Corn  Frozen  1.6889   
75                            Potatoes, french fries  Frozen  1.9018   
28                                       Cauliflower  Frozen  2.3974   
65                                    Mustard greens  Frozen  2.4324   

   RetailPriceUnit   Yield  CupEquivalentSize CupEquivalentUnit  \
24       per pound  0.6500             0.3197            pounds   
44       per pound  0.6500             0.3527            pounds   
42       per pound  0.6500             0.2976            pounds   
55       per pound  0.6500             0.3748            pounds   
39       per pound  0.6500             0.3858            pounds   
..             ...     ...                ...               ...   
61       per pound  0.9700             0.3105            pounds   
36       per pound  0.9630             0.3638            pounds   
75       per pound  0.7760             0.3417            pounds   
28       per pound  0.9304             0.2756            pounds   
65       per pound  0.7478             0.3307            pounds   

    CupEquivalentPrice  
24              0.5892  
44              0.7119  
42              0.5170  
55              0.8525  
39              0.6875  
..                 ...  
61              0.6895  
36              0.6380  
75              0.8374  
28              0.7101  
65              1.0756  

[93 rows x 8 columns]

In [14]:
df[5:10] #slice the rows 5-10

Veggie    Form      RP RetailPriceUnit   Yield  CupEquivalentSize  \
5    Asparagus  Frozen  6.8212       per pound  1.0335             0.3968   
6     Avocados   Fresh  2.6737       per pound  0.7408             0.3197   
7        Beets  Canned  1.1431       per pound  0.6500             0.3748   
8  Black beans  Canned  1.2387       per pound  0.6500             0.3858   
9  Black beans   Dried  1.5250       per pound  2.4692             0.3858   

  CupEquivalentUnit  CupEquivalentPrice  
5            pounds              2.6191  
6            pounds              1.1538  
7            pounds              0.6591  
8            pounds              0.7352  
9            pounds              0.2383

In [15]:
df[df['RP'] > 3] # select rows by values

Veggie    Form      RP RetailPriceUnit   Yield  \
2                  Artichoke  Canned  3.4498       per pound  0.6500   
4                  Asparagus  Canned  3.4328       per pound  0.6500   
5                  Asparagus  Frozen  6.8212       per pound  1.0335   
13            Broccoli heads   Fresh  3.0820       per pound  0.7800   
26       Cauliflower florets   Fresh  3.8022       per pound  0.9702   
31            Collard greens   Fresh  3.0881       per pound  1.1600   
47                      Kale   Fresh  3.4338       per pound  1.0500   
54  Lettuce, romaine, hearts   Fresh  3.2802       per pound  0.8500   
62          Mushrooms, whole   Fresh  3.8929       per pound  0.9700   
63         Mushrooms, sliced   Fresh  3.9224       per pound  1.0000   
68                      Okra   Fresh  5.1567       per pound  0.7695   
70                    Olives  Canned  5.3496       per pound  1.0000   
80           Spinach, boiled   Fresh  4.1214       per pound  0.7700   
81        Spinach, eaten raw   Fresh  4.1214       per pound  1.0000   
85  Tomatoes, grape & cherry   Fresh  3.8729       per pound  0.9100   

    CupEquivalentSize CupEquivalentUnit  CupEquivalentPrice  
2              0.3858            pounds              2.0476  
4              0.3968            pounds              2.0958  
5              0.3968            pounds              2.6191  
13             0.3417            pounds              1.3502  
26             0.2756            pounds              1.0800  
31             0.2866            pounds              0.7630  
47             0.2866            pounds              0.9373  
54             0.2094            pounds              0.8082  
62             0.1543            pounds              0.6194  
63             0.1543            pounds              0.6053  
68             0.3527            pounds              2.3639  
70             0.2976            pounds              1.5922  
80             0.3307            pounds              1.7700  
81             0.1543            pounds              0.6360  
85             0.3748            pounds              1.5950

- I selected and displayed the "Veggie" column which tells us there are 93 types of vegetables recorded in this data
- I then divided/sorted the data by form the veggies are in such as canned, frozen or dried
- Next sliced the rows 5-10 to only show those specfic rows
- Lastly, I sorted the rows by only showing veggies with a retail price less than 3 (> 3)

## Group & Visualize the Data